Tests/FluentAssertions.Specs/Collections/GenericCollectionAssertionsSpecs.cs

```diff
@@ -1931,5 +1931,237 @@ public void When_inspectors_count_does_not_equal_asserting_collection_length_it_
             act.Should().Throw<XunitException>().WithMessage("*because we want to test the failure*");
         }
         #endregion
+
+        #region Satisfy (Predicates)
+
+        [Fact]
+        public void When_collection_element_at_each_position_matches_predicate_at_same_position_Satisfy_should_not_throw()
+        {
+            // Arrange
+            var collection = new int[] { 1, 2, 3 };
+
+            // Act
+            Action act = () => collection.Should().Satisfy(
+                element => element == 1,
+                element => element == 2,
+                element => element == 3);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_collection_element_at_each_position_matches_predicate_at_reverse_position_Satisfy_should_not_throw()
+        {
+            // Arrange
+            var collection = new int[] { 1, 2, 3 };
+
+            // Act
+            Action act = () => collection.Should().Satisfy(
+                element => element == 3,
+                element => element == 2,
+                element => element == 1);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_one_element_does_not_have_matching_predicate_Satisfy_should_throw()
+        {
+            // Arrange
+            var collection = new int[] { 1, 2 };
+
+            // Act
+            Action act = () => collection.Should().Satisfy(
+                element => element == 3,
+                element => element == 1,
+                element => element == 2);
+
+            // Assert
+            act.Should().Throw<XunitException>();
```

Comment:
🤔  I would at least assert a part of the exception message (using wildcards) to ensure it the missing predicate is reported.

Reply:
I believe the assertion of the missing predicate is handled in `When_Satisfy_fails_then_failure_message_must_contain_predicates_without_matching_elements_and_elements_without_matching_predicates`.

Reply:
I added the validation. Won't harm for sure ;)